.buildkite/scripts/integration_tests.sh

```diff
@@ -5,10 +5,20 @@ WORKSPACE="$(pwd)"
 TMP_FOLDER_TEMPLATE_BASE="tmp.elastic-package"
 
 cleanup() {
+    local error_code=$?
+
+    if [ $error_code != 0 ] ; then
+        if [ -f ${GOOGLE_APPLICATION_CREDENTIALS+x} ]; then
+             google_cloud_logout_active_account
+        fi
+    fi
+
     echo "Deleting temporal files..."
     cd ${WORKSPACE}
     rm -rf "${TMP_FOLDER_TEMPLATE_BASE}.*"
     echo "Done."
+
+    exit $error_code
 }
 trap cleanup EXIT
```

Comment:
I didn't want to move this function to `tooling.sh` to avoid more refactors in this PR.

Each pipeline (script) has its own cleanup and the one from `release` pipeline is different.

Reply:
I tried to take into account this by using this [parameter expansion](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02) `${var+x}` 

```shell
 if [ -n "${GOOGLE_APPLICATION_CREDENTIALS+x}" ]; then
```

In that case, if the variable is not defined the logout is not executed.

I think that it could be a good approach, since that variable is set by the `google_cloud_auth` just after the login command

https://github.com/elastic/elastic-package/blob/f1b90c7227f090da3c0a4dfb6c6711d67084d90a/.buildkite/scripts/tooling.sh#L23-L25

In [this commit](https://github.com/elastic/elastic-package/pull/1487/commits/34c03753e623f17c35fc5011c11293e3e36b17c6) I added two conditions that the variable is defined and the file pointing that variable exists. Maybe it can be just leave that the variable is defined. If the variable is defined, run the logout function

@sharbuz WDYT ?